### PR TITLE
W3DRopeDraw

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/client/draw/w3dropedraw.h [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    void Toss_Segments();
+
+    static Module *Friend_New_Module_Instance(Thing *thing, ModuleData const *module_data);
+
+private:
+    struct SegInfo
+    {
+        Line3DClass *m_line1;
+        Line3DClass *m_line2;
+        float m_sinAngle;
```
Comment (tomsons26) [inline src/game/client/draw/w3dropedraw.h]: these are swapped

Comment (jonwil) [reply src/game/client/draw/w3dropedraw.h]: fixed.

File: src/game/client/draw/w3dropedraw.cpp [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        float f1 = GameMath::Sin(m_wobbleAngle) * m_wobbleAmplitude;
+        const Coord3D *pos = Get_Drawable()->Get_Position();
+        Vector3 v1(pos->x, pos->y, pos->z + m_unk2);
+        float f2 = m_ropeLength / m_segments.size();
+
+        for (auto segment = m_segments.begin(); segment != m_segments.end(); segment++) {
+            Vector3 v2(f1 * segment->m_cosAngle + pos->x, f1 * segment->m_sinAngle + pos->y, v1.Z - f2);
+
+            if (segment->m_line1 != nullptr) {
+                segment->m_line1->Reset(v1, v2);
```
Comment (tomsons26) [inline src/game/client/draw/w3dropedraw.cpp]: start, end

Comment (jonwil) [reply src/game/client/draw/w3dropedraw.cpp]: fixed.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**